a2.cpp

```
@@ -0,0 +1,74 @@
+
+public:
+    Trie() {
+        root = new TrieNode();
+    }
+
+    // Insert a word into the Trie
+    void insert(string word) {
+        TrieNode* current = root;
+        for (char c : word) {
+            if (current->children.find(c) == current->children.end()) {
+                current->children[c] = new TrieNode();
```

cppcheck (2.10-2)

[performance] Searching before insertion is not necessary. Instead of 'current->children[c]=new TrieNode()' consider using 'current->children.try_emplace(c, new TrieNode());'.

(stlFindInsert)